### DISCUSSION TITLE
How to define a router in a library

### DISCUSSION THREAD

**Discussion by pct-cclausen** [trpc/trpc#5061] [Q&A] [answered]
In our nx monorepo we have a bunch of libraries that define APIs.
I'd like to define them as trpc routers so they can be used in multiple server-app projects in the monorepo.

From the documentation it seems I need to initTRPC.create({...]}) and then take the router from there to define new routers.
But that initTRPC is meant to be in the server, is it not?

How can my library define a router without the need for that?
Or is it ok to just do initTRPC.context.create in every lib and the server?

Thanks for the help, tRPC looks amazingly useful to us.

**Comment by KATT** [comment, marked as answer]
Define the initTRPC root object only once in its own package (but it'll work if you don't as well)

**Reply by pct-cclausen** [reply]
hmmm, ok

Thanks for your great work
